packages/client/src/lib.js

```diff
@@ -94,15 +104,40 @@ class NFTStorage {
   }
   /**
    * @param {API.Service} service
-   * @param {Blob} blob
-   * @returns {Promise<API.CIDString>}
+   * @param {Blob|CarReader} carReader
+   * @returns {Promise<API.CIDString | undefined>}
    */
-  static async storeCar({ endpoint, token }, blob) {
-    const car =
-      blob.type !== 'application/car'
-        ? blob.slice(0, blob.size, 'application/car') // TODO: test this in safari.
-        : blob
-    return NFTStorage.storeBlob({ endpoint, token }, car)
+  static async storeCar({ endpoint, token }, carReader) {
+    // See if this is a blob
+    if (carReader instanceof Blob) {
+      const blob = carReader
+      carReader = await CarReader.fromIterable(toIterable(blob.stream()))
+    }
+
+    // const roots = await carReader.getRoots()
+    // TODO: Should we throw if multiple roots?
+
+    const targetSize = 960 * 960 * 100 // chunk to ~96MB CARs
+    // const targetSize = 960 * 960 * 4
+    const splitter = new TreewalkCarSplitter(carReader, targetSize)
+
+    const upload = parallelMap(
+      3,
+      async (/** @type {AsyncIterable<Uint8Array>} */ car) => {
+        const carParts = await all(car)
+        const carFile = new Blob(carParts, {
+          type: 'application/car',
+        })
+        console.log('store blob')
+        const res = await NFTStorage.storeBlob({ endpoint, token }, carFile)
+        console.log('stored blob')
+        return res
+      }
+    )
+
+    const res = await all(upload(splitter.cars()))
+
+    return res && res[0]
```

Comment:
Lets not import another module for this!

```suggestion
  let cid
  for await (const res of upload(splitter.cars())) {
    cid = res
  }
  return cid
```

Reply:
Just pointing out that we need to do this both here and for the async iterator from the spiller itself. I removed the module for now, but can put it back if you reconsider